### PR TITLE
🐛 Fix signup issues #282

### LINKED ISSUE #282: 🐛 vistor login page hangs after reading QR code
Uhh ooo... 75% sure this isn't a me issue, but you never know 😊
screen shot 2018-03-18 at 15 34 45

### PR DESCRIPTION
**Pull request by astroash**
- rename state.gender
- remove this.this.state
- rename smsing stuffs

fixes #282

### CODE REVIEW

**Comment by RogeredBacon** [comment]
I'd be up for getting this merged ASAP if you've fixed stuff as I'm working on these two pages

**Comment by astroash** [comment]
@RogeredBacon I was just writing some tests and adding `Joi` in.

File: react-backend/routes/__tests__/qr.generator.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const test = require('tape');
+const request = require('supertest');
+const jwt = require('jsonwebtoken');
+const createApp = require('../../app');
+const { getConfig } = require('../../../config');
+
+const config = getConfig(process.env.NODE_ENV);
+
+test('POST /api//qr/generator | Correct payload', t => {
```
Comment (eliascodes) [inline react-backend/routes/__tests__/qr.generator.test.js]: Double `//` and in all the calls to `supertest`. Does express ignore the double slash? How are the tests working? 😕

File: react-backend/routes/qr_generator.js [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
 const hash = require('../functions/hash');
 const sendQrCode = require('../functions/qr_send');
 
-router.post('/', (req, res, next) => {
+const schema = {
+  body: {
+    formSender: Joi.string().required(), 
+    formPhone: Joi.number().required(), 
+    formGender: Joi.string().required(), 
+    formYear: Joi.number().required(),
```
Comment (eliascodes) [inline react-backend/routes/qr_generator.js]: We can easily put a min/max on this. Let's say min of 120 years ago.

File: react-backend/routes/qr_generator.js [hunk 3]
```diff
@@ -1,17 +1,31 @@
...
+const { validate } = require('../shared/middleware');
 const qrcodemaker = require('../functions/qrcodemaker');
 const hash = require('../functions/hash');
 const sendQrCode = require('../functions/qr_send');
 
-router.post('/', (req, res, next) => {
+const schema = {
+  body: {
+    formSender: Joi.string().required(), 
+    formPhone: Joi.number().required(),
```
Comment (eliascodes) [inline react-backend/routes/qr_generator.js]: What about phone numbers that begin with 0? Or that start with `++`? Or have dashes?

**Pull request merged**